### PR TITLE
20 1 workspace members

### PR DESCRIPTION
**Pull request by Artem672**

### CODE REVIEW

File: Provis.Core/Services/WorkspaceService.cs [hunk 1]
```diff
@@ -265,5 +265,30 @@ public async Task<WorkspaceInfoDTO> GetWorkspaceInfoAsync(int workspId, string u
 
             return workspace;
         }
+        public async Task<List<WorkspaceMemberDTO>> GetWorkspaceMembersAsync(int workspaceId)
+        {
+            var workspace = await _workspaceRepository
+                .Query()
+                .FirstOrDefaultAsync(w => w.Id == workspaceId);
```
Comment (pashunskyi) [inline Provis.Core/Services/WorkspaceService.cs]: maby better do "_workspaceRepository.GetByKeyAsync(workspaceId)"?

Comment (Artem672) [reply Provis.Core/Services/WorkspaceService.cs]: Great suggestion, fixed

File: Provis.Core/Services/WorkspaceService.cs [hunk 2]
```diff
@@ -265,5 +265,28 @@ public async Task<WorkspaceInfoDTO> GetWorkspaceInfoAsync(int workspId, string u
 
             return workspace;
         }
+        public async Task<List<WorkspaceMemberDTO>> GetWorkspaceMembersAsync(int workspaceId)
+        {
+            var workspace = await _workspaceRepository.GetByKeyAsync(workspaceId);
+            if (workspace == null)
+                throw new HttpException
+                    (System.Net.HttpStatusCode.NotFound,
```
Comment (YevheniiVovk) [inline Provis.Core/Services/WorkspaceService.cs]: Let's make consistency formatting

Comment (Artem672) [reply Provis.Core/Services/WorkspaceService.cs]: Fixed

**Review by YevheniiVovk — APPROVED** [review]

**Review by AndriyKovalets — APPROVED** [review]

File: Provis.Core/Services/WorkspaceService.cs [hunk 3]
```diff
@@ -265,5 +265,30 @@ public async Task<WorkspaceInfoDTO> GetWorkspaceInfoAsync(int workspId, string u
...
+                {
+                    Id = o.UserId,
+                    Role = o.Role.Name,
+                    UserName = o.User.UserName
+                })
+                .OrderBy(o => o.UserName)
+                .ToListAsync();
+
+            return workspaceMembers;
+        }
```
Comment (AndriyKovalets) [inline Provis.Core/Services/WorkspaceService.cs]: Why don't you check to see if the user is a member of the workspace? Or any user can get a list of members from any workspace?

Comment (Artem672) [reply Provis.Core/Services/WorkspaceService.cs]: Great suggestion, fixed

File: Provis.Core/Services/WorkspaceService.cs [hunk 4]
```diff
@@ -257,13 +257,44 @@ public async Task<WorkspaceInfoDTO> GetWorkspaceInfoAsync(int workspId, string u
...
+                    UserName = o.User.UserName
+                })
+                .OrderBy(o => o.UserName)
+                .ToListAsync();
+
+            if(workspaceMembers.FirstOrDefault(o => o.Id == userId) == null)
+            {
+                throw new HttpException(System.Net.HttpStatusCode.Forbidden,
+                    "You don't have access to this workspace!");
+            }
```
Comment (AndriyKovalets) [inline Provis.Core/Services/WorkspaceService.cs]: Delete this if and add policy in controller

Comment (Artem672) [reply Provis.Core/Services/WorkspaceService.cs]: Fixed

**Review by AndriyKovalets — APPROVED** [review]

**Pull request merged**